requests/exceptions.py

```diff
@@ -100,6 +100,8 @@ class StreamConsumedError(RequestException, TypeError):
 class RetryError(RequestException):
     """Custom retries logic failed"""
 
+class RedirectError(RequestException):
```

Comment:
Eh, let's just be specific and call it an "UnrewindableBodyError". Specificity never hurt anyone.